test/apps/iodemo/io_demo.cc

```diff
@@ -2916,10 +2955,10 @@ static int parse_args(int argc, char **argv, options_t *test_opts)
     }
 
     while (optind < argc) {
-        test_opts->servers.push_back(argv[optind++]);
+        servers.push_back(argv[optind++]);
```

Comment:
maybe instead of using local server array, we can call something like add_servers(argv[optind++]) for each argument. and add each server including repeat count if needed

Reply:
can we round-robin the src addresses between all servers that we created (either directly or by repeat count)?
IMO it will make the code simpler and more generic - remove the dependency between test_opts->src_addrs.size() and servers.size()

Reply:
>
>can we round-robin the src addresses between all servers that we created (either directly or by repeat count)?
>
I didn't understand it well.
1. ```src_addrs``` is used one by one to connect with the server.
2. suppose there're two source addresses ```src_ip0, src_ip1``` and there're two servers ```server0:port0, server1:port1:2```.
    1) The current logic is to use ```src_ip0``` to connect with ```server0:port0``` and use ```src_ip1``` to connect with ```server1:port1``` for ```2``` times.
    2) Do you mean that it should ```src_ip0``` to connect with ```server0:port0``` and use ```src_ip1``` to connect with ```server1:port1``` for the first time, then use ```src_ip0``` to connect with ```server1:port1``` for the second time?

Reply:
in case (2), there are actually 3 servers: server0:port0, server1:port1, server1:port1
yes, i would suggest in this case we use src_addrs in the following order: 
server0:port0 - src_ip0
server1:port1 - src_ip1
server1:port1 - src_ip0

Reply:
it's up to the user to specify right combination of server and src_addr

Reply:
Got it. I'll simplify the code here and update this PR implementation on weekend time.

Reply:
The code is simplified a lot after following this idea to make the design.